tests/test_path_analysis.py

```diff
@@ -44,85 +59,118 @@ def target(data):
-                               res_space = 3,
-                               node_threshold = 0.1,
-                               edge_threshold = 0.1,
-                               normalize = False)
-    metapath = pa.reorder_graph(metapath, data[0])
-    metapath = nx.to_numpy_matrix(metapath)
-    return metapath
+def combinations(data):
+    return pa.get_combinations(res_id = data[0],
+                               res_space = 3)
+
+@pytest.fixture
+def all_shortest_paths(example_metapath, id):
+    return pa.get_all_shortest_paths(graph = example_metapath, 
+                                     res_id = id, 
+                                     res_space = 1)
+
+@pytest.fixture
+def graph_from_paths(all_shortest_paths):
+    return pa.get_graph_from_paths(all_shortest_paths)
+
+@pytest.fixture
+def filtered_graph(graph_from_paths):
+    return pa.filter_graph(graph_from_paths, 0.3, 0.3)
+
+@pytest.fixture
+def normalized_graph(graph_from_paths):
+    return pa.normalize_graph(graph_from_paths)
 
 @pytest.fixture
-def metapath_norm(data):
-    metapath = pa.get_metapath(graph = data[1],
+def metapath(example_metapath, id):
+    return pa.get_metapath(graph = example_metapath,
+                                      res_id = id,
+                                      res_space = 1,
+                                      node_threshold = 0.3,
+                                      edge_threshold = 0.3,
+                                      normalize = False)
+
+@pytest.fixture
+def reordered_graph(data):
+    graph = pa.get_metapath(graph = data[1],
                                res_id = data[0],
                                res_space = 3,
                                node_threshold = 0.1,
                                edge_threshold = 0.1,
-                               normalize = True)
-    metapath = pa.reorder_graph(metapath, data[0])
-    metapath = nx.to_numpy_matrix(metapath)
-    return metapath
-
-# Test shortest paths
-def test_shortest_path(shortest_table, ref_name):
-    ref_csv = []
-    with open(ref_name['shortest_csv'], "r") as f:
-        for line in f:
-            # remove white space and split line
-            li, s, t, l, w1, w2 = line.rstrip().split('\t')
-            # change to correct format
-            line = (li.split(','), s, t, int(l), float(w1), float(w2))
-            ref_csv.append(line)
-    assert shortest_table == ref_csv
-
-def test_shortest_path_graph(shortest_path_graph, ref_name):
-    ref_graph = np.loadtxt(ref_name['shortest_dat'])
-    graph = nx.to_numpy_matrix(shortest_path_graph)
-    assert_equal(graph, ref_graph)
+                               normalize = False)
+    graph = pa.reorder_graph(graph, data[0])
+    return graph
 
-# Test simple paths
-def test_all_path(all_table, ref_name):
+@pytest.fixture
+def metapath_matrix(reordered_graph):
+    matrix = nx.to_numpy_matrix(reordered_graph)
+    return matrix
+
+# Test path functions
+def test_convert_input_to_list(source):
```

Comment:
You could add comments to the functions (like the function description in triple quotes if you want to do it for all or just normal comments if you want to do it sparsely) to give some details about the tests.